src/resizing.jl

```diff
@@ -156,34 +156,58 @@ end
 # imresize
 imresize(original::AbstractArray, dim1, dimN...) = imresize(original, (dim1,dimN...))
 
-function imresize{T,N}(original::AbstractArray{T,N}, short_size::NTuple)
+function imresize(original::AbstractArray, short_size::Tuple)
     len_short = length(short_size)
-    new_size = ntuple(i -> (i > len_short ? size(original,i) : short_size[i]), N)
+    new_size = ntuple(i -> (i > len_short ? odims(original, i, short_size) : short_size[i]), N)
```

Comment:
If you don't mind I have a follow up question on "runtime vs compile time". I think I am mixing up terminology here and I would like to learn how to, well, avoid that.

- Based on my intuition I would assume that without the `N`, the JIT compilation would fail once the function is called. I.e. it could never produce any machine code that could be executed. (I understand, though, that in this particular example the function wasn't called because of the lack of any test).

- Since I still think the above is true (please do correct me if it isn't), I am guessing I am simply using the term "compile time" incorrectly.

Where do you, as a Julia developer, draw the line between compile time and runtime? Do you only count the execution of the Julia code that defines the function as compile time (like pasting a function in the REPL and pressing enter)? Is the JIT compilation thought of as a run-time event, and consequently if it fails it is called a runtime error?